pkg/controller/validate/validate.go

```diff
@@ -59,3 +74,50 @@ func allAreNil(values ...interface{}) bool {
 func moreThanOneIsNonNil(values ...interface{}) bool {
 	return getNonNilCount(values...) > 1
 }
+
+func instanceSizeForAdvancedDeployment(replicationSpecs []*mdbv1.AdvancedReplicationSpec) error {
+	var instanceSize string
+	err := errors.New("instance size must be the same for all nodes in all regions and across all replication specs for advanced deployment ")
+
+	for _, replicationSpec := range replicationSpecs {
+		for _, regionSpec := range replicationSpec.RegionConfigs {
+			if instanceSize == "" {
+				instanceSize = regionSpec.ElectableSpecs.InstanceSize
+			}
+
+			if regionSpec.ElectableSpecs.InstanceSize != instanceSize {
+				return err
+			}
+
+			if regionSpec.ReadOnlySpecs.InstanceSize != instanceSize {
+				return err
+			}
+
+			if regionSpec.AnalyticsSpecs.InstanceSize != instanceSize {
```

Comment:
@helderjs `Specs` needs to be checked if they are nil. They are not always filled. This is the reason why helm tests are failing.